components/prism-elm.js

```diff
@@ -0,0 +1,35 @@
+Prism.languages.elm = {
+	comment: {
+		pattern: /(^|[^-!#$%*+=?&@|~.:<>^\\\/])(--[^-!#$%*+=?&@|~.:<>^\\\/].*|{-[\s\S]*?-})/m,
+		lookbehind: true
+	},
+	char: /'([^\\']|\\([abfnrtv\\"'&]|\^[A-Z@[\]\^_]|NUL|SOH|STX|ETX|EOT|ENQ|ACK|BEL|BS|HT|LF|VT|FF|CR|SO|SI|DLE|DC1|DC2|DC3|DC4|NAK|SYN|ETB|CAN|EM|SUB|ESC|FS|GS|RS|US|SP|DEL|\d+|o[0-7]+|x[0-9a-fA-F]+))'/,
+	string: {
+		pattern: /"([^\\"]|\\([abfnrtv\\"'&]|\^[A-Z@[\]\^_]|NUL|SOH|STX|ETX|EOT|ENQ|ACK|BEL|BS|HT|LF|VT|FF|CR|SO|SI|DLE|DC1|DC2|DC3|DC4|NAK|SYN|ETB|CAN|EM|SUB|ESC|FS|GS|RS|US|SP|DEL|\d+|o[0-7]+|x[0-9a-fA-F]+)|\\\s+\\)*"/,
+		greedy: true
+	},
+	import_statement: {
+		// The imported or hidden names are not included in this import
+		// statement. This is because we want to highlight those exactly like
+		// we do for the names in the program.
+		pattern: /(\r?\n|\r|^)\s*import\s+([A-Z][_a-zA-Z0-9]*)(\.[A-Z][_a-zA-Z0-9]*)*(\s+as\s+([A-Z][_a-zA-Z0-9]*)(\.[A-Z][_a-zA-Z0-9]*)*)?(\s+exposing\s+)?/m,
+		inside: {
+			keyword: /\b(import|as|exposing)\b/
+		}
+	},
+	keyword: /\b(alias|as|case|else|exposing|if|in|infixl|infixr|let|module|of|then|type)\b/,
+	// These are builtin variables only. Constructors are highlighted later as a constant.
+	builtin: /\b(abs|acos|always|asin|atan|atan2|ceiling|clamp|compare|cos|curry|degrees|e|flip|floor|fromPolar|identity|isInfinite|isNaN|logBase|max|min|negate|never|not|pi|radians|rem|round|sin|sqrt|tan|toFloat|toPolar|toString|truncate|turns|uncurry|xor)\b/,
+	// decimal integers and floating point numbers | octal integers | hexadecimal integers
+	number: /\b(\d+(\.\d+)?(e[+-]?\d+)?|0x[0-9a-f]+)\b/i,
+	// Most of this is needed because of the meaning of a single '.'.
+	// If it stands alone freely, it is the function composition.
+	// It may also be a separator between a module name and an identifier => no
+	// operator. If it comes together with other special characters it is an
+	// operator too.
+	operator: /\s\.\s|[-!#$%*+=?&@|~.:<>^\\\/]*\.[-!#$%*+=?&@|~.:<>^\\\/]+|[-!#$%*+=?&@|~.:<>^\\\/]+\.[-!#$%*+=?&@|~.:<>^\\\/]*|[-!#$%*+=?&@|~:<>^\\\/]+|`([A-Z][_a-zA-Z0-9']*\.)*[_a-z][_a-zA-Z0-9']*`/,
```

Comment:
- Occurrences of `[_a-zA-Z0-9']` should be replaced with `[\w']`.
- I might be wrong, but couldn't the whole regexp be rewritten as this one?
(single dot | special char (including dot) appearing two or more times | special char alone (not including dot) | backtick stuff)
``/\s\.\s|[-!#$%*+=?&@|~:<>^\\\/.]{2,}|[-!#$%*+=?&@|~:<>^\\\/]|`([A-Z][\w']*\.)*[_a-z][\w']*`/``

Reply:
Yep. I'm gonna have a look at the characters the actual Elm parser allows, which should allow me to scrap a few. I included the backtick stuff for compatibility with Elm 0.17 but I think there's no real point to doing that, especially with 0.19 coming close to release.

I'll simplify, thanks :)